src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -1230,18 +1230,21 @@ private IEnumerable<TypeInferenceInfo> InferTypeInForEachStatement(ForEachStatem
                     return SpecializedCollections.EmptyEnumerable<TypeInferenceInfo>();
                 }
 
+
+                var enumerableType = forEachStatementSyntax.AwaitKeyword == default
+                    ? this.Compilation.GetSpecialType(SpecialType.System_Collections_Generic_IEnumerable_T)
+                    : this.Compilation.GetTypeByMetadataName(typeof(IAsyncEnumerable<>).FullName);
```

Comment:
Cannot make a SpecialType for IAsyncEnumerable because it's not in (Co-located core types)[https://github.com/dotnet/roslyn/blob/master/docs/compilers/Co-located%20core%20types.md]

Reply:
have to check enumerableType for null after this.

Reply:
@CyrusNajmabadi Done